### PR TITLE
feat(mobile): add missing bottom sheet to FW update

### PR DESCRIPTION
**Pull request by Nodonisko**
## Description

Added missing bottom sheet to warn that you should have seed ready and also added may be stucked bottom sheet.

Also refactored FW update progres indicator to reflect new design changes.

## Related Issue

Resolve part #15584 

## Screenshots:
Simulator Screenshot - iPhone 16 Pro - 2024-12-09 at 00 35 25

https://github.com/user-attachments/assets/0cb85f79-d16a-409f-b119-d789439f8445

https://github.com/user-attachments/assets/558c5281-e5d2-448f-a32f-cb1f2364a1be

### CODE REVIEW

File: suite-native/atoms/src/NumberedListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+type NumberedListItemProps = {
+    children: ReactNode;
+    variant?: TypographyStyle;
+    color?: Color;
+    number: number;
+};
+
+export const NumberedListItem = ({ children, variant, color, number }: NumberedListItemProps) => (
+    <HStack>
+        <Box style={{ minWidth: 16 }}>
```
Comment (PeKne) [inline suite-native/atoms/src/NumberedListItem.tsx]: Please no inline styles

File: suite-native/firmware/src/components/MayBeStuckedBottomSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { BottomSheet, Box, Button, NumberedListItem, Text, VStack } from '@suite-native/atoms';
+import { Translation } from '@suite-native/intl';
+
+type MayBeStuckedBottomSheetProps = {
+    isOpened: boolean;
+    onClose: () => void;
+};
+
+export const MayBeStuckedBottomSheet = ({ isOpened, onClose }: MayBeStuckedBottomSheetProps) => {
+    const [visiblePart, setVisiblePart] = useState<number>(1);
```
Comment (PeKne) [inline suite-native/firmware/src/components/MayBeStuckedBottomSheet.tsx]: nit:
```suggestion
    const [visiblePart, setVisiblePart] = useState<1 | 2>(1);
```

**Review by PeKne — APPROVED** [review]

File: suite-native/atoms/src/NumberedListItem.tsx (hunk 1 above)
Comment (Nodonisko) [reply suite-native/atoms/src/NumberedListItem.tsx]: fixed

File: suite-native/firmware/src/components/MayBeStuckedBottomSheet.tsx (hunk 2 above)
Comment (Nodonisko) [reply suite-native/firmware/src/components/MayBeStuckedBottomSheet.tsx]: fixed

**Review by PeKne — APPROVED** [review]

**Pull request merged**